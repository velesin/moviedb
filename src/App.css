.App {
    box-sizing: border-box;
    width: 1400px;
    min-height: 100vh;
}