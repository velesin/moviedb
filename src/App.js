import React, { Component } from 'react';
import { search } from './services/movies.service';
import { SearchBox } from './search/SearchBox';
import { SearchResultList } from './search/SearchResultList';

class App extends Component {
    state = {
        movies: []
    }

    doSearch = async query => {
        const { results: movies } = await search(query);
        this.setState({ movies });
    }
    
    render() {
        return (
            <div className="Search">
                <SearchBox onQuerySubmit={ this.doSearch } />
                <SearchResultList movies={ this.state.movies }/>
            </div>            
        );
    }    
}

export default App;
