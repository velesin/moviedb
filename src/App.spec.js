jest.mock('./services/movies.service');

import React from 'react';
import { shallow } from 'enzyme';

import { search } from './services/movies.service'
import App from './App';

let app;

beforeEach(() => {
    app = shallow(<App />);    
});

it('renders correct initial page layout', () => {
    expect(app.getElement()).toMatchSnapshot();
});

describe('when user submits a query via the search box', () => {    
    const QUERY = 'QUERY';
    const FOUND_MOVIES = 'FOUND_MOVIES';
    
    beforeEach(async () => {
        search.mockReturnValue({ results: FOUND_MOVIES });
        search.mockClear();

        await app.find('SearchBox').simulate('querySubmit', QUERY);
        app.update();
    });

    it('sends the query to the API', () => {
        expect(search.mock.calls[0][0]).toEqual(QUERY);
    });

    it('updates the result list with the movies received from the API', () => {
        expect(app.find('SearchResultList').prop('movies')).toEqual(FOUND_MOVIES);
    });
});
