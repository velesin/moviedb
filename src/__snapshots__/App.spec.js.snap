// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct initial page layout 1`] = `
<div
  className="App"
>
  <SearchBox
    onQuerySubmit={[Function]}
  />
  <SearchResultList
    movies={Array []}
  />
</div>
`;
